apps/sensors_test/src/main.c

```diff
@@ -413,10 +410,10 @@ config_sensor(void)
 
     dev = (struct os_dev *) os_dev_open("light0", OS_TIMEOUT_NEVER, NULL);
     assert(dev != NULL);
-    rc = tsl2561_init(dev, NULL);
-    if (rc) {
-        os_dev_close(dev);
-        goto err;
+
+    if (dev->od_flags & OS_DEV_F_STATUS_READY) {
+        rc= SYS_ENODEV;
+        console_printf("config_sensor %s init failed\n", dev->od_name);
```

Comment:
There would be more changes to come. I will explain them in detail together. git follows a 50/72 limit and hence fitting everything in 50 characters is not possible. Adding a high level title is exactly what I have done. If you expand the title you will see additional comments below.